### PR TITLE
Sketch Graph Level IR and Support Multiple Engine

### PR DESCRIPTION
**Pull request by livingshade**
Currently we only support linear connection like "fault->logging" 
Example usage:
`python main.py -e "fault->logging->fault"`
Now we also support multiple instance of same engine.

### CODE REVIEW

File: compiler/main.py [hunk 1]
```diff
@@ -59,4 +62,47 @@
...
         f.write("\n".join(ctx.process_code))
 
-    compiler.generate(engine_name, ctx, args.phoenix_dir)
+    compiler.finalize(engine, ctx, mrpc_dir)
+
+
+if __name__ == "__main__":
+
+    # Parse command line arguments
+    parser = argparse.ArgumentParser()
```
Comment (Romero027) [inline compiler/main.py:71]: Can you explain the arguments and their usage in the README? (Especially the mrpc dir option - the default one seems too specific to one platform)

**Review by Romero027 — CHANGES_REQUESTED** [review]
one comment. Otherwise looks good to me

**Comment by Kristoff-starling** [comment]
LGTM except a tiny detail. To make the e2e demo work automatically, maybe we should generate another script to execute the mrpc commands (in the next PR)

**Comment by Romero027** [comment]
lgtm, thanks Banruo.

**Pull request merged**